lib/rubocop/cop/lint/uri_regexp.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop identifies places where `URI.regexp`
+      # can be replaced by `URI::Parser.new.make_regexp`.
+      #
+      # @example
+      #   # bad
+      #   URI.regexp("http://example.com")
+      #
+      #   # good
+      #   URI::Parser.new.make_regexp("http://example.com")
+      #
+      class UriRegexp < Cop
+        MSG = 'Use `URI::Parser.new.make_regexp%s` instead of `URI.regexp%s`.'
+              .freeze
+
+        def_node_matcher :uri_regexp_with_argument?, <<-PATTERN
+          (send
+            (const nil :URI) :regexp
```

Comment:
It does not match `::URI.regexp`

Reply:
Thanks for the review. I missed that viewpoint. I fixed by 6bdf4e6.